androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/SurveyDetailScreen.kt

```diff
@@ -22,36 +25,20 @@ private const val ImageScaleAnimationDurationInMillis = 700
 @Composable
 fun SurveyDetailScreen(
     homeViewModel: HomeViewModel = getViewModel(),
+    surveyDetailViewModel: SurveyDetailViewModel = getViewModel(),
     surveyId: String,
     onBackClick: () -> Unit,
 ) {
-    val viewState by homeViewModel.viewState.collectAsStateWithLifecycle()
-    val surveyUiModel = viewState.surveys.find { it.id == surveyId }
+    val homeViewState by homeViewModel.viewState.collectAsStateWithLifecycle()
+    val surveyDetailViewState by surveyDetailViewModel.viewSate.collectAsStateWithLifecycle()
     var shouldShowStartContent by remember { mutableStateOf(false) }
     var shouldShowSurveyQuestionContent by remember { mutableStateOf(false) }
     var imageScale by remember { mutableStateOf(InitialImageScale) }
     val coroutineScope = rememberCoroutineScope()
-    // TODO: use property from the ViewModel in the integrate task
-    var surveyQuestionUiModels by remember { mutableStateOf<List<SurveyQuestionUiModel>>(emptyList()) }
+    val surveyUiModel = homeViewState.surveys.find { it.id == surveyId }
 
-    // TODO: simulating network latency. Remove in the integrate task.
     LaunchedEffect(Unit) {
-        delay(1000)
-        // Call API and fetch the questions
-        surveyQuestionUiModels = List(5) {
-            SurveyQuestionUiModel(
-                id = it.plus(1).toString(),
-                text = "How fulfilled did you feel during this WFH period?",
-                displayType = DisplayType.DROPDOWN,
-                imageUrl = surveyUiModel?.largeImageUrl.orEmpty(),
-                answerUiModels = List(5) {
-                    AnswerUiModel(
-                        id = (it + 1).toString(),
-                        text = "Text ${it + 1}"
-                    )
-                }
-            )
-        }
+        surveyDetailViewModel.fetchSurveyDetail(surveyId)
     }
 
     LaunchedEffect(Unit) {
```

Comment:
Lets combine both `LaunchedEffect` into one.

Reply:
Resolved in https://github.com/nimblehq/avishek-phong-kmm-ic/pull/134/commits/82c9c5c4e33b934ed828ecdcd77f166808f91442